File: client-ios/SerenadaCore/Sources/Models/CallStats.swift
import Foundation

public struct CallStats: Equatable {
    public var bitrate: Double?
    public var packetLoss: Double?
    public var jitter: Double?
    public var codec: String?
    public var iceCandidatePair: String?
    public var roundTripTime: Double?
    public var audioRxKbps: Double?
    public var audioTxKbps: Double?
    public var videoRxKbps: Double?
    public var videoTxKbps: Double?
    public var videoFps: Double?
    public var videoResolution: String?
    public var updatedAtMs: Int64 = 0

    public static let empty = CallStats()

    public init() {}

    public init(from realtimeStats: RealtimeCallStats) {
        self.bitrate = realtimeStats.availableOutgoingKbps
        self.packetLoss = realtimeStats.videoRxPacketLossPct ?? realtimeStats.audioRxPacketLossPct
        self.jitter = realtimeStats.audioJitterMs
        self.roundTripTime = realtimeStats.rttMs
        self.audioRxKbps = realtimeStats.audioRxKbps
        self.audioTxKbps = realtimeStats.audioTxKbps
        self.videoRxKbps = realtimeStats.videoRxKbps
        self.videoTxKbps = realtimeStats.videoTxKbps
        self.videoFps = realtimeStats.videoFps
        self.videoResolution = realtimeStats.videoResolution
        self.iceCandidatePair = realtimeStats.transportPath
        self.updatedAtMs = realtimeStats.updatedAtMs
    }
}

File: client-android/serenada-core/src/main/java/app/serenada/core/CallStats.kt
package app.serenada.core

import app.serenada.core.call.RealtimeCallStats

/**
 * Aggregated call statistics exposed to SDK consumers.
 * Populated from WebRTC getStats() periodically during an active call.
 */
data class CallStats(
    val bitrate: Double? = null,
    val packetLoss: Double? = null,
    val jitter: Double? = null,
    val codec: String? = null,
    val iceCandidatePair: String? = null,
    val roundTripTime: Double? = null,
    val audioRxKbps: Double? = null,
    val audioTxKbps: Double? = null,
    val videoRxKbps: Double? = null,
    val videoTxKbps: Double? = null,
    val videoFps: Double? = null,
    val videoResolution: String? = null,
    val realtimeStats: RealtimeCallStats? = null,
    val updatedAtMs: Long = 0L,
)
